.pre-commit-config.yaml

```diff
@@ -1,68 +1,38 @@
 ---
 ci:
-    autoupdate_schedule: quarterly
-
+  autoupdate_schedule: quarterly
 repos:
-    - repo: https://github.com/pre-commit/pre-commit-hooks
-      rev: v4.6.0
-      hooks:
-          - id: end-of-file-fixer
-          - id: trailing-whitespace
-            exclude: miscellaneous/structures/SiO2.xyz
-          - id: check-yaml
-          - id: check-added-large-files
-            args: [--maxkb=6000]
-    - repo: https://github.com/pycqa/isort
-      rev: 5.13.2
-      hooks:
-          - id: isort
-            args: [--profile, black, --filter-files]
-    - repo: https://github.com/PyCQA/autoflake
-      rev: v2.3.1
-      hooks:
-          - id: autoflake
-    - repo: https://github.com/asottile/pyupgrade
-      rev: v3.16.0
-      hooks:
-          - id: pyupgrade
-            args: [--py38-plus]
-    - repo: https://github.com/psf/black
-      rev: 24.4.2
-      hooks:
-          - id: black
-            language_version: python3 # Should be a command that runs python3.6+
-    - repo: https://github.com/PyCQA/flake8
-      rev: 7.1.0
-      hooks:
-          - id: flake8
-            args: [--count, --show-source, --statistics]
-            additional_dependencies:
-                - flake8-bugbear
-                - flake8-builtins
-                - flake8-comprehensions
-                - flake8-debugger
-                - flake8-logging-format
-                - pep8-naming
-                - pyflakes
-                - tryceratops
-    # - repo: https://github.com/pre-commit/mirrors-mypy
-    #   rev: v1.5.1
-    #   hooks:
-    #       - id: mypy
-    #         additional_dependencies:
-    #             - types-click-spinner
-    #             - types-requests
-    #             - types-tabulate
-    #             - types-toml
-    - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
-      rev: 0.2.3
-      hooks:
-          - id: yamlfmt
-    - repo: https://github.com/asottile/setup-cfg-fmt
-      rev: v2.5.0
-      hooks:
-          - id: setup-cfg-fmt
-    - repo: https://github.com/kynan/nbstripout
-      rev: 0.7.1
-      hooks:
-          - id: nbstripout
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-yaml
+      - id: check-added-large-files
+        args: [--maxkb=6000]
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.7.4
+    hooks:
+      # Ruff fix
+      - id: ruff
+        types_or: [python, pyi]
+        args: [--fix]
+        name: ruff (fix)
+      # Ruff formatter
+      - id: ruff-format
+        types_or: [python, pyi]
+        name: ruff (format)
+  - repo: https://github.com/google/yamlfmt
+    rev: v0.14.0
+    hooks:
+      - id: yamlfmt
+        name: YAML (format)
+        types: [yaml]
+  - repo: https://github.com/asottile/setup-cfg-fmt
+    rev: v2.7.0
+    hooks:
+      - id: setup-cfg-fmt
```

Comment:
Maybe delete this one too? I am not sure we are using it at all.

Reply:
Indeed. We don't have `setup.cfg` files. I'll go ahead, remove that and merge this.